packages/pipeline/src/Pipeline.tsx

```diff
@@ -0,0 +1,285 @@
+import React, {
+  forwardRef,
+  createRef,
+  useState,
+  Ref,
+  ReactElement,
+  ReactNode,
+} from 'react';
+
+import PropTypes from 'prop-types';
+import findLast from 'lodash/findLast';
+
+import { uiColors } from '@leafygreen-ui/palette';
+import Tooltip from '@leafygreen-ui/tooltip';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { useElementNode, useMutationObserver } from '@leafygreen-ui/hooks';
+
+import Stage from './Stage';
+import Counter from './Counter';
+
+import {
+  getPipelineCounterTooltip,
+  isStageElement,
+  isElementOverflowed,
+} from './utils';
+
+import {
+  getRootStyle,
+  Size,
+  Variant,
+  layout,
+  colors,
+  pipelineAttr,
+  pipelineStages,
+  stageChevronAttr,
+} from './styles';
+
+interface StateForStyles {
+  hasHiddenStages: boolean;
+  size: Size;
+  variant: Variant;
+}
+
+interface PipelineProps {
+  /**
+   * Content that will appear inside of the Pipeline component.
+   */
+  children: ReactNode;
+
+  /**
+   * Optional className prop to apply to Pipeline content container.
+   */
+  className?: string;
+
+  /**
+   * Alter the rendered size of the component
+   */
+  size: Size;
+
+  /**
+   * Alter the visual appearance of the component
+   */
+  variant: Variant;
+}
+
+const getBaseStyle = ({ size, variant }: StateForStyles): string =>
+  cx(
+    getRootStyle({ size, variant }),
+    css`
+      counter-reset: hiddenCount;
+      flex-direction: row;
+    `,
+  );
+
+const getPipelineStyle = ({ size, variant }: StateForStyles): string => {
+  const { minWidth } = layout[size];
+
+  return cx(
+    getRootStyle({ size, variant }),
+    css`
+      flex-grow: 1;
+      flex-shrink: 1;
+      flex-basis: 100%;
+      flex-wrap: wrap;
+      overflow: hidden;
+      list-style: none;
+      padding: 0;
+      margin: 0;
+      min-width: ${minWidth}px;
+    `,
+  );
+};
+
+export const lastVisibleClassName =
```

Comment:
The `:last-of-type` selector allows you to target the last occurrence of an element within its container. This will not work for this use case because what I actually need to find is the last *visible* occurrence of an element within its container (ie: the last element before the parent container clips the others when content overflows).

If I use the `createDataProp` it will clash with another usage of the `createDataProp` on this element (because the `createDataProp` doesn't allow me to set the key of the resulting attribute, only the value). As far as I am aware, `data-` attributes should have unique names, which would make this not an option?